Compile error on linux

Hi when I try to manually compile the program on Manjaro Linux Kernel 6.4.2-3 I get an error.


### The error.     

When I compile the program  I get the following ouput:                                                                                                                    
"warning: dependency (hidapi) specified without providing a local path, Git repository, or version to use. This will be considered
 an error in future versions
   Compiling annepro2_tools v0.1.0 (My-project-path)
error[E0599]: no method named \`usage\` found for reference \`&DeviceInfo\` in the current scope
   --> src/annepro2.rs:111:17
    |
111 |             dev.usage(),
    |                 ^^^^^ private field, not a method

error[E0599]: no method named \`usage_page\` found for reference \`&DeviceInfo\` in the current scope
   --> src/annepro2.rs:112:17
    |
112 |             dev.usage_page(),
    |                 ^^^^^^^^^^ private field, not a method

For more information about this error, try `rustc --explain E0599`.
error: could not compile `annepro2_tools` (bin "annepro2_tools") due to 2 previous errors"


I found that the newer versions of hidapi does not have those methods. It has private fields instead.


### The solution.

In the `Cargo.toml` I added a specific earlier version of hidapi. Since i could see that some of the project was 3 years old I took a 3 year old version. 

Under `[dependencies]`
Put: 
`hidapi = {version = "1.2.3", default-features = false, features = ["linux-static-libusb"]}`

Instead of 

`hidapi = {default-features = false, features = ["linux-static-libusb"]}`

It now compiles.

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -7,6 +7,6 @@ edition = "2018"
 # See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
 
 [dependencies]
-hidapi = { version = "2.4.1", default-features = false, features = ["linux-static-libusb"]}
-pretty-hex = "0.3.0"
+hidapi = { version = "2.6.0", default-features = false, features = ["linux-static-libusb"] }
+pretty-hex = "0.4.1"
 structopt = "0.3.26"

--- a/src/annepro2.rs
+++ b/src/annepro2.rs
@@ -105,13 +105,11 @@ pub fn flash_firmware<R: std::io::Read>(
 fn fetch_devices(api: &HidApi) -> (Vec<&hidapi::DeviceInfo>, Option<&hidapi::DeviceInfo>) {
     for dev in api.device_list() {
         println!(
-            "HID Dev: {:04x}:{:04x} usage #: {:02x} usage_page #: {:04x} {}",
+            "HID Dev: {:04x}:{:04x} {}",
             dev.vendor_id(),
             dev.product_id(),
-            dev.usage(),
-            dev.usage_page(),
             dev.product_string()
-                .map(|it| format!("({:})", it.replace("\n", " - ")))
+                .map(|it| format!("({:})", it.replace('\n', " - ")))
                 .unwrap_or_default()
         );
     }

--- a/src/annepro2.rs
+++ b/src/annepro2.rs
@@ -1,5 +1,5 @@
 use hidapi::{HidApi, HidDevice, HidResult};
-use std::{intrinsics::transmute, thread, time::Duration};
+use std::{thread, time::Duration};
 
 const ANNEPRO2_VID: u16 = 0x04d9;
 
@@ -179,7 +179,7 @@ pub fn write_chunk(
     chunk: &[u8],
 ) -> HidResult<()> {
     let mut buffer: Vec<u8> = vec![L2Command::FW as u8, KeyCommand::IapWirteMemory as u8];
-    let addr_slice: [u8; 4] = unsafe { transmute(addr.to_le()) };
+    let addr_slice: [u8; 4] = addr.to_le_bytes();
     buffer.extend_from_slice(&addr_slice);
     buffer.extend_from_slice(chunk);
     write_to_target(handle, target, &buffer).map(|_| ())
@@ -187,7 +187,7 @@ pub fn write_chunk(
 
 pub fn erase_device(handle: &HidDevice, target: AP2Target, addr: u32) -> HidResult<()> {
     let mut buffer: Vec<u8> = vec![L2Command::FW as u8, KeyCommand::IapEraseMemory as u8];
-    let addr_slice: [u8; 4] = unsafe { transmute(addr.to_le()) };
+    let addr_slice: [u8; 4] = addr.to_le_bytes();
     buffer.extend_from_slice(&addr_slice);
 
     write_to_target(handle, target, &buffer)?;

